.ci_support/update_environment.py

```diff
@@ -0,0 +1,68 @@
+import json
+import re
+import sys
+
+import yaml
+
+environment_file = '.ci_support/environment.yml'
+name_mapping_file = '.ci_support/pypi_vs_conda_names.json'
+
+
+class EnvironmentUpdater:
+    def __init__(self, package_name, from_version, to_version):
+        """
+        Updates the version of a package in the conda environment file.
+
+        Parameters:
+            package_name: Name of the package to update as available on PyPI
+            from_version: Version the package is before the update
+            to_version: Version to which the package should be updated
+        """
+        self.from_version = from_version
+        self.to_version = to_version
+        with open(name_mapping_file, 'r') as f:
+            self._name_conversion_dict = json.load(f)
+
+        with open(environment_file, 'r') as f:
+            self.environment = yaml.safe_load(f)
+
+        self.package_name = self._convert_package_name(package_name)
+
+    def _convert_package_name(self, name):
+        try:
+            result = self._name_conversion_dict[name]
+        except KeyError:
+            result = name
```

Comment:
You can check if `name` is in `self._name_conversion_dict.keys()` then you do not need the `try` and `except` part.